File: 2025/05 May/24 Sum of all substrings of a number/main.py
# Link: https://www.geeksforgeeks.org/problems/sum-of-all-substrings-of-a-number-1587115621/1


class Solution:
    def sumSubstrings(self, s):
        sum = 0
        mf = 1
        for i in range(len(s) - 1, -1, -1):
            sum = sum + (int(s[i])) * (i + 1) * mf
            mf = mf * 10 + 1

        return sum


if __name__ == "__main__":
    obj = Solution()
    s = "6759"
    print(obj.sumSubstrings(s))

    s = "421"
    print(obj.sumSubstrings(s))

File: 2025/05 May/24 Sum of all substrings of a number/main.cpp
// Link: https://www.geeksforgeeks.org/problems/sum-of-all-substrings-of-a-number-1587115621/1

#include <iostream>
using namespace std;

class Solution
{
public:
    int sumSubstrings(string &s)
    {
        long long int sum = 0;

        long long int mf = 1;
        for (int i = s.size() - 1; i >= 0; i--)
        {
            sum += (s[i] - '0') * (i + 1) * mf;
            mf = mf * 10 + 1;
        }

        return sum;
    }
};

int main()
{
    Solution solution;
    string s = "6759";
    cout << "Sum of all substrings of " << s << " is: " << solution.sumSubstrings(s) << endl;

    s = "421";
    cout << "Sum of all substrings of " << s << " is: " << solution.sumSubstrings(s) << endl;
    return 0;
}